[ISSUE] Error when updating Unity Catalog table

### Configuration
```hcl
resource "databricks_table" "this" {
  name               = "test"

  ## uncomment next line to force update of the table on next apply
  # owner = "unity_admin"

  catalog_name       = "main"
  schema_name        = "default"
  
  table_type         = "MANAGED"
  data_source_format = "DELTA"

  column {
    name      = "id"
    position  = 0
    type_name = "INT"
    type_text = "int"
    type_json = "{\"name\":\"id\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}"
  }
}
```

### Expected Behavior
Updating a table should work

### Actual Behavior
The provider returns the following error when trying to update the table:

**`error: cannot update table: UpdateTable invalid full_name_arg`**
```
databricks_table.this: Modifying... [id=main.default.test]
2022-03-10T23:54:50.634+0100 [INFO]  Starting apply for databricks_table.this
2022-03-10T23:54:50.634+0100 [DEBUG] databricks_table.this: applying the planned Update change
2022-03-10T23:54:50.635+0100 [INFO]  provider.terraform-provider-databricks_v0.5.2: Using directly configured basic authentication: timestamp=2022-03-10T23:54:50.634+0100
2022-03-10T23:54:50.635+0100 [INFO]  provider.terraform-provider-databricks_v0.5.2: Configured basic auth: host=https://***REDACTED***.cloud.databricks.com/, username=***REDACTED***, password=***REDACTED***, account_id=***REDACTED***: timestamp=2022-03-10T23:54:50.635+0100
2022-03-10T23:54:50.635+0100 [DEBUG] provider.terraform-provider-databricks_v0.5.2: PATCH /api/2.0/unity-catalog/tables/test {
2022-03-10T23:54:51.257+0100 [DEBUG] provider.terraform-provider-databricks_v0.5.2: 400 Bad Request {
2022-03-10T23:54:51.257+0100 [WARN]  provider.terraform-provider-databricks_v0.5.2: /api/2.0/unity-catalog/tables/test:400 - UpdateTable invalid full_name_arg.: timestamp=2022-03-10T23:54:51.257+0100
2022-03-10T23:54:51.257+0100 [WARN]  provider.terraform-provider-databricks_v0.5.2: /api/2.0/unity-catalog/tables/test:400 - UpdateTable invalid full_name_arg.: timestamp=2022-03-10T23:54:51.257+0100
2022-03-10T23:54:51.260+0100 [ERROR] vertex "databricks_table.this" error: cannot update table: UpdateTable invalid full_name_arg.
│   with databricks_table.this,
│   on main.tf line 65, in resource "databricks_table" "this":
│   65: resource "databricks_table" "this" {
path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.2/darwin_arm64/terraform-provider-databricks_v0.5.2 pid=39212
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. perform any modification on the table, for example specifying an owner
3. `terraform apply` 

### Terraform and provider versions

Databricks provider 0.5.2
Terraform 1.1.0

### Debug Output
(provided above)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@
 * Use correct HTTP verb for modifying `databricks_permissions` on `databricks_sql_endpoint` entities. Authorized user, assumingly part of `admins` group, is no longer sending `CAN_MANAGE` permission in the HTTP PUT request ([#1163](https://github.com/databrickslabs/terraform-provider-databricks/issues/1163)).
 * Added diff suppression for `min_num_clusters` field in `databricks_sql_endpoint` ([#1172](https://github.com/databrickslabs/terraform-provider-databricks/pull/1172)).
 * Added special case for handling `Cannot access cluster that was terminated or unpinned more than 30 days ago` error in `databricks_cluster` as an indication of resource removed on the platform side ([#1177](https://github.com/databrickslabs/terraform-provider-databricks/issues/1177)).
+* Fixed updating of `databricks_table` resources ([#1175](https://github.com/databrickslabs/terraform-provider-databricks/issues/1175)).
 
 Updated dependency versions:
 

--- a/catalog/resource_table.go
+++ b/catalog/resource_table.go
@@ -72,7 +72,7 @@ func (a TablesAPI) getTable(name string) (ti TableInfo, err error) {
 }
 
 func (a TablesAPI) updateTable(ti TableInfo) error {
-	return a.client.Patch(a.context, "/unity-catalog/tables/"+ti.Name, ti)
+	return a.client.Patch(a.context, "/unity-catalog/tables/"+ti.FullName(), ti)
 }
 
 func (a TablesAPI) deleteTable(name string) error {

